### PR TITLE
Show correct waypoint type for generated waypoints. 

### LINKED ISSUE #12892: Wrong waypoint type for generated waypoints when editing
### Describe your problem!

When editing a generated waypoint, the waypoint type of the spinner shows "parking"

### How to reproduce?

Add a waypoint with calculated coordinates, which has a variable with a range.
Create waypoints from that.

Screenshot_20220406-213748_cgeo

Edit one of the auto-generated waypoints.

### Actual result after these steps?

In the activity, the type for the waypoint is shown as "parking"
Screenshot_20220406-213730_cgeo

### Expected result after these steps?

In the activity, the type for the waypoint should be shown as "generated"

### Reproducible

Yes

### c:geo Version

2022.04.01-NB-511d6f2

### System information

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by murggel**
## Description

Show correct waypoint type for generated waypoints. 
Use the underlaying adapter to retrieve possible waypoint-types

## Related issues

fix #12892

### CODE REVIEW

File: main/src/cgeo/geocaching/EditWaypointActivity.java [hunk 1]
```diff
@@ -333,29 +333,29 @@ public void onNothingSelected(final AdapterView<?> parent) {
...
-    private int getDefaultWaypointType() {
+    private int getDefaultWaypointTypePosition(final ArrayAdapter<WaypointType> wpAdapter) {
         // potentially restore saved instance state
         if (waypointTypeSelectorPosition >= 0) {
             return waypointTypeSelectorPosition;
         }
 
         // when editing, use the existing type
         if (waypoint != null) {
-            return POSSIBLE_WAYPOINT_TYPES.indexOf(waypoint.getWaypointType());
```
Comment (eddiemuc) [inline main/src/cgeo/geocaching/EditWaypointActivity.java:348]: Where else is this "POSSIBLE_WAYPOINT_TYPES" constant used? Can it be deleted now? If not: might this change cause conflicts with places where it is still in use?

File: main/src/cgeo/geocaching/EditWaypointActivity.java [hunk 2]
```diff
@@ -333,29 +333,29 @@ public void onNothingSelected(final AdapterView<?> parent) {
...
+            return wpAdapter.getPosition(waypoint.getWaypointType());
```
Comment (eddiemuc) [inline main/src/cgeo/geocaching/EditWaypointActivity.java:348]: What should happen if (for whatever reason) the waypoint type is not found in the adapter? Returning -1 ok?

**Review by eddiemuc — COMMENTED** [review]
Some questions

File: main/src/cgeo/geocaching/EditWaypointActivity.java (hunk 1 above)
Comment (murggel) [reply main/src/cgeo/geocaching/EditWaypointActivity.java:348]: The array is used to initialize the spinner, depending on wheter it is a generated waypoint POSSIBLE_WAYPOINT_TYPES or POSSIBLE_WAYPOINT_TYPES_WITH_GENERATED is used.

File: main/src/cgeo/geocaching/EditWaypointActivity.java (hunk 2 above)
Comment (murggel) [reply main/src/cgeo/geocaching/EditWaypointActivity.java:348]: Good question - will show what the old return-value was and was the new one is.

Comment (eddiemuc) [reply main/src/cgeo/geocaching/EditWaypointActivity.java:348]: ok, then it is finer

File: main/src/cgeo/geocaching/EditWaypointActivity.java (hunk 1 above)
Comment (eddiemuc) [reply main/src/cgeo/geocaching/EditWaypointActivity.java:348]: ok

**Pull request merged**